src/daft-scan/src/scan_task_iters.rs

```diff
@@ -111,3 +116,109 @@ impl Iterator for MergeByFileSize {
         }
     }
 }
+
+pub fn split_by_row_groups(
+    scan_tasks: BoxScanTaskIter,
+    max_tasks: usize,
+    split_threshold_bytes: usize,
+    min_size_bytes: usize,
+) -> BoxScanTaskIter {
+    let mut scan_tasks = itertools::peek_nth(scan_tasks);
+
+    // only split if we have a small amount of files
+    if scan_tasks.peek_nth(max_tasks).is_some() {
+        return Box::new(scan_tasks);
+    }
+
+    Box::new(
+        scan_tasks
+            .map(move |t| -> DaftResult<BoxScanTaskIter> {
+                let t = t?;
+
+                /* Only split parquet tasks if they:
+                   - have one source
+                   - use native storage config
+                   - have no specified chunk spec or number of rows
+                   - have size past split threshold
+                */
+                match (
+                    t.file_format_config.as_ref(),
+                    t.storage_config.as_ref(),
+                    &t.sources[..],
+                    t.sources.get(0).map(DataFileSource::get_chunk_spec),
+                    t.pushdowns.limit,
+                ) {
+                    (
+                        FileFormatConfig::Parquet(_),
+                        StorageConfig::Native(_),
+                        [source],
+                        Some(None),
+                        None,
+                    ) if source
+                        .get_size_bytes()
+                        .map_or(true, |s| s > split_threshold_bytes as u64) =>
+                    {
+                        let (io_runtime, io_client) =
+                            t.storage_config.get_io_client_and_runtime()?;
+
+                        let path = source.get_path();
+
+                        let io_stats =
+                            IOStatsContext::new(format!("split_by_row_groups for {:#?}", path));
+
+                        let runtime_handle = io_runtime.handle();
+
+                        let file = runtime_handle.block_on(read_parquet_metadata(
+                            path,
+                            io_client,
+                            Some(io_stats),
+                        ))?;
+
+                        let mut new_tasks: Vec<DaftResult<ScanTaskRef>> = Vec::new();
+                        let mut curr_row_groups = Vec::new();
+                        let mut curr_size_bytes = 0;
+
+                        for (i, rg) in file.row_groups.iter().enumerate() {
+                            curr_row_groups.push(i as i64);
+                            curr_size_bytes += rg.compressed_size();
```

Comment:
I did this because I believe the original `size_bytes` for these parquet scan tasks are just the file sizes, which are compressed as well, so this would match the behavior.

https://github.com/Eventual-Inc/Daft/blob/69df969036a15093f407e62eda5c748e8c12f503/src/daft-scan/src/glob.rs#L256